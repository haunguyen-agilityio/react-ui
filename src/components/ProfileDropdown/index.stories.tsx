import React from 'react'

import { ComponentStory, ComponentMeta } from '@storybook/react'

import ProfileDropdown from '@components/ProfileDropdown'

export default {
  title: 'Components/Dropdown',
  component: ProfileDropdown,
} as ComponentMeta<typeof ProfileDropdown>

const Template: ComponentStory<typeof ProfileDropdown> = (args) => <ProfileDropdown {...args} />

export const Primary = Template.bind({})
Primary.args = {
  isDisabled: false,
  isDisableGutters: false,
}

export const Secondary = Template.bind({})
Secondary.args = {
  isDisabled: false,
  isDisableGutters: true,
}

export const Disable = Template.bind({})
Disable.args = {
  isDisabled: true,
  isDisableGutters: true,
}
