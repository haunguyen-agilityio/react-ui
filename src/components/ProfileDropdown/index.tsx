// Libraries
import AccordionDetails from '@mui/material/AccordionDetails'
import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
import { Paper } from '@mui/material'
import Typography from '@mui/material/Typography'

// Styles
import {
  DropdownStyles,
  StyledAccordion,
  StyledAccordionSummary,
  StyledMenuItem,
  StyledMenuList,
  StyledLink,
} from '../ProfileDropdown/styles'

// Themes
import theme from '../../themes'

interface IProfileDropdownProps {
  isDisableGutters: boolean
  isDisabled: boolean
}

const ProfileDropdown = ({
  isDisableGutters = false,
  isDisabled = false,
}: IProfileDropdownProps) => {
  return (
    <DropdownStyles>
      <StyledAccordion disableGutters={isDisableGutters} disabled={isDisabled}>
        <StyledAccordionSummary
          expandIcon={<ExpandMoreIcon />}
          aria-controls='panel1a-content'
          id='panel1a-header'
        >
          <Typography fontSize={theme.spacing(1.6)} align={'inherit'} variant={'h4'}>
            Food Bar Inc
          </Typography>
        </StyledAccordionSummary>
        <AccordionDetails>
          <Typography fontSize={theme.spacing(1.6)} align={'inherit'} variant={'h4'}>
            Hello, world
          </Typography>
        </AccordionDetails>
      </StyledAccordion>
      <Paper>
        <StyledMenuList>
          <StyledMenuItem>
            {/*TODO: Add Link from next/Link*/}
            <StyledLink variant={'h4'}>Address</StyledLink>
          </StyledMenuItem>
          <StyledMenuItem>
            <StyledLink variant={'h4'}>Documents</StyledLink>
          </StyledMenuItem>
          <StyledMenuItem>
            <StyledLink variant={'h4'}>Profile Settings</StyledLink>
          </StyledMenuItem>
          <StyledMenuItem>
            <StyledLink variant={'h4'}>Log out</StyledLink>
          </StyledMenuItem>
        </StyledMenuList>
      </Paper>
    </DropdownStyles>
  )
}

export default ProfileDropdown
