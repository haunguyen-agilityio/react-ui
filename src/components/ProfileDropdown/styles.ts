// Libraries
import { styled } from '@mui/system'
import Accordion from '@mui/material/Accordion'
import AccordionSummary from '@mui/material/AccordionSummary'
import { MenuItem, MenuList, Link } from '@mui/material'

// Themes
import theme from '@themes'

export const DropdownStyles = styled('div')({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  width: '100%',
})

export const StyledAccordion = styled(Accordion)({
  width: theme.spacing(26.8),
  padding: theme.spacing(1.6),
  fontSize: theme.spacing(1.6),
})

export const StyledAccordionSummary = styled(AccordionSummary)({
  border: `${theme.spacing(0.1)} solid ${theme.palette.grey[300]}`,
})

export const StyledMenuItem = styled(MenuItem)({
  borderTop: `${theme.spacing(0.1)} solid ${theme.palette.grey[300]}`,
  padding: theme.spacing(1.6),
})

export const StyledMenuList = styled(MenuList)({
  width: theme.spacing(30),
  padding: theme.spacing(0),
})

export const StyledLink = styled(Link)({
  textDecoration: 'none',
  color: theme.palette.text.primary,
  fontSize: theme.spacing(1.6),
})
