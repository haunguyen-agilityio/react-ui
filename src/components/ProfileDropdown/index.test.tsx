import { screen, render } from '@testing-library/react'
import * as renderer from 'react-test-renderer'
import ProfileDropdown from '../ProfileDropdown'

describe('Dropdown', () => {
  it('Component [ProfileDropdown] matches DOM Snapshot', () => {
    const component = renderer.create(
      <ProfileDropdown isDisableGutters={false} isDisabled={false} />,
    )
    expect(component).toMatchSnapshot()
  })

  it('Component [Dropdown] should render correctly content', () => {
    render(<ProfileDropdown isDisableGutters={false} isDisabled={false} />)

    expect(screen.getByText('Hello, world')).toBeInTheDocument()
    expect(screen.getByText('Address')).toBeInTheDocument()
    expect(screen.getByText('Documents')).toBeInTheDocument()
    expect(screen.getByText('Profile Settings')).toBeInTheDocument()
    expect(screen.getByText('Log out')).toBeInTheDocument()
  })
})
