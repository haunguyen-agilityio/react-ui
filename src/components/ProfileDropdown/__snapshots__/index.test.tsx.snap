// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component [ProfileDropdown] matches DOM Snapshot 1`] = `
<div
  className="css-hbm2t4"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1hl5cpj-MuiPaper-root-MuiAccordion-root"
  >
    <div
      aria-controls="panel1a-content"
      aria-expanded={false}
      className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-oiohog-MuiButtonBase-root-MuiAccordionSummary-root"
      id="panel1a-header"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 css-36gbox-MuiTypography-root"
        >
          Food Bar Inc
        </h4>
      </div>
      <div
        className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="panel1a-header"
            className="MuiAccordion-region"
            id="panel1a-content"
            role="region"
          >
            <div
              className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
            >
              <h4
                className="MuiTypography-root MuiTypography-h4 css-36gbox-MuiTypography-root"
              >
                Hello, world
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
  >
    <ul
      className="MuiList-root MuiList-padding css-1l3vh8n-MuiList-root"
      onKeyDown={[Function]}
      role="menu"
      tabIndex={-1}
    >
      <li
        className="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-2ca55l-MuiButtonBase-root-MuiMenuItem-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <a
          className="MuiTypography-root MuiTypography-h4 MuiLink-root MuiLink-underlineAlways css-9wqgu2-MuiTypography-root-MuiLink-root"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Address
        </a>
      </li>
      <li
        className="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-2ca55l-MuiButtonBase-root-MuiMenuItem-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <a
          className="MuiTypography-root MuiTypography-h4 MuiLink-root MuiLink-underlineAlways css-9wqgu2-MuiTypography-root-MuiLink-root"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Documents
        </a>
      </li>
      <li
        className="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-2ca55l-MuiButtonBase-root-MuiMenuItem-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <a
          className="MuiTypography-root MuiTypography-h4 MuiLink-root MuiLink-underlineAlways css-9wqgu2-MuiTypography-root-MuiLink-root"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Profile Settings
        </a>
      </li>
      <li
        className="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-2ca55l-MuiButtonBase-root-MuiMenuItem-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="menuitem"
        tabIndex={-1}
      >
        <a
          className="MuiTypography-root MuiTypography-h4 MuiLink-root MuiLink-underlineAlways css-9wqgu2-MuiTypography-root-MuiLink-root"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Log out
        </a>
      </li>
    </ul>
  </div>
</div>
`;
